Controller/ResourceController.php

```diff
@@ -39,11 +46,99 @@ public function __construct(
         $this->registry = $registry;
     }
 
-    public function resourceAction($repositoryName, $path)
+    public function getResourceAction($repositoryName, $path)
     {
         $repository = $this->registry->get($repositoryName);
         $resource = $repository->get('/'.$path);
 
+        return $this->createResponse($resource);
+    }
+
+    /**
+     * There are two different changes that can currently be done on a cmf resource:
+     *
+     * - move
+     * - rename
+     *
+     * changing payload properties isn't supported yet.
+     *
+     * @param string $repositoryName
+     * @param string $path
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function patchResourceAction($repositoryName, $path, Request $request)
+    {
+        $repository = $this->registry->get($repositoryName);
+        $this->failOnNotEditable($repository, $repositoryName);
+
+        $path = '/'.ltrim($path, '/');
+
+        $requestContent = json_decode($request->getContent(), true);
+        if (!$requestContent) {
+            return $this->badRequestResponse('Only JSON request bodies are supported.');
+        }
+
+        foreach ($requestContent as $action) {
+            switch ($action['operation']) {
```

Comment:
Did you read http://williamdurand.fr/2014/02/14/please-do-not-patch-like-an-idiot/ ? 

 > You can use whatever format you want as [description of changes], as far as its semantics is well-defined. That is why using PATCH to send updated values only is not suitable.

 > RFC 6902 defines a JSON document structure for expressing a sequence of operations to apply to a JSON document, suitable for use with the PATCH method. Here is how it looks like:

 > ```
 > [
 >     { "op": "test", "path": "/a/b/c", "value": "foo" },
 >     { "op": "remove", "path": "/a/b/c" },
 >     { "op": "add", "path": "/a/b/c", "value": [ "foo", "bar" ] },
 >     { "op": "replace", "path": "/a/b/c", "value": 42 },
 >     { "op": "move", "from": "/a/b/c", "path": "/a/b/d" },
 >     { "op": "copy", "from": "/a/b/d", "path": "/a/b/e" }
 > ]
 > ```